code/modules/mob/living/carbon/xenomorph/castes/warrior/abilities_warrior.dm

```diff
@@ -449,19 +464,15 @@
 	Shake(4, 4, 2 SECONDS)
 	return TRUE
 
-/mob/living/punch_act(mob/living/carbon/xenomorph/X, damage, target_zone)
-	var/punch_description = "powerful"
-	var/stagger_stacks = 3
-	var/slowdown_stacks = 3
-
+/mob/living/punch_act(mob/living/carbon/xenomorph/warrior/X, damage, target_zone, push = TRUE, punch_description = "powerful", stagger_stacks = 3, slowdown_stacks = 3)
 	if(pulledby == X) //If we're being grappled by the Warrior punching us, it's gonna do extra damage and debuffs; combolicious
 		damage *= 2
 		slowdown_stacks *= 2
 		stagger_stacks *= 2
 		ParalyzeNoChain(0.5 SECONDS)
 		punch_description = "devastating"
 
-	if(iscarbon(src) && !isxeno(src)) //Xenos don't have limbs really
+	if(iscarbon(src))
```

Comment:
fix this please, this should be a overwrirtten proc on mob/living/carbon, rather that a shitty istype(src)